.github/workflows/ci-release.yml

```diff
@@ -34,35 +38,57 @@ jobs:
           SONATYPE_USER: ${{ secrets.SONATYPE_USER }}
           SONATYPE_TOKEN: ${{ secrets.SONATYPE_TOKEN }}
           CI: true
-        run: ./gradlew --parallel --max-workers=4 -PreleaseBuild=true clean check && ./gradlew -PreleaseBuild=true publish
+        run: |
+          # Build arguments to feed to the single gradlew publish command
+          if [ "${{ matrix.java }}" = "8" ]; then
+            FIRST_GRADLE_TARGETS=" clean check"
+            SECOND_GRADLE_TARGETS=" publish"
+          else
+            FIRST_GRADLE_TARGETS=""
+            SECOND_GRADLE_TARGETS=""
+
+            # Execute the printJavaTargetCompatibility task to get the java target compatibility for each subproject
+            # and extract the projects that require jdk9+.
+            while read -r line
+            do
+              javaTarget=$(echo "$line" | sed -e 's/^version: \(.*\) name:.*/\1/g')
+              if [ "$javaTarget" = "1.9" ] || [ "$javaTarget" = "1.10" ] || [ "$javaTarget" -gt "8" ] 2> /dev/null
+              then
+                currDir=$(echo "$line" | sed -e 's/^version:.* name: \(.*\)$/\1/g')
+                FIRST_GRADLE_TARGETS="$FIRST_GRADLE_TARGETS :$currDir:clean :$currDir:check"
+                SECOND_GRADLE_TARGETS="$SECOND_GRADLE_TARGETS :$currDir:publish"
+              fi
+            done < <(./gradlew printJavaTargetCompatibility)
+          fi
+
+          # Execute the gradlew command to publish the build
+          ./gradlew --parallel --max-workers=4 -PreleaseBuild=true$FIRST_GRADLE_TARGETS && ./gradlew -PreleaseBuild=true$SECOND_GRADLE_TARGETS
```

Comment:
I expected we will just manually list modules that require JDK9+, but this automated approach is even better, thanks!

Reply:
I started with manual listing, but too many manual listings laying around and we will likely forget to update it in the future.